src/rewrites.lisp

```diff
@@ -117,112 +117,145 @@ Example patterns:
                              :test car=-fn))))))
 
 (defun default-car-match-fn (a b)
-  (if (eql a b)
-      nil
-      'no-match))
+  (unless (eql a b)
+    'no-match))
```

Comment:
don't remove NIL if the value is meaningful. in this case it is since a boolean is being returned.

use WHEN / UNLESS for side effects only